### ISSUE TITLE
locks don't timeout as advertised

### ISSUE THREAD

**Issue by sgendler-stem** [gruntwork-io/terragrunt#371]
I have a template with a lock that seems to be perma-locked.  For several days, it has just blocked on "Acquiring state lock. This may take a few moments..."

However, I have NEVER run anything without -lock-timeout=20m in the command.  However, a command did get interrupted by network loss on the day when the lock got locked.  I'm guessing acquiring the lock and setting the timeout are somehow not part of the same atomic operation?  I'm using the following in my top-level terraform.tfvars file:

```
  remote_state {
    backend = "s3"

    config {
      bucket = "<redacted>"
      key    = "${path_relative_to_include()}/terraform.tfstate"
      region = "us-west-2"
      encrypt = true
      dynamodb_table = "terraform-lock-table"
    }

  terraform {
    extra_arguments "common_vars" {
      commands = ["${get_terraform_commands_that_need_vars()}"]

      optional_var_files = [
        "${get_tfvars_dir()}/${find_in_parent_folders("account.tfvars", "skip-account-if-does-not-exist")}",
        "${get_tfvars_dir()}/${find_in_parent_folders("region.tfvars", "skip-region-if-does-not-exist")}",
        "${get_tfvars_dir()}/${find_in_parent_folders("env.tfvars", "skip-env-if-does-not-exist")}",
        "${get_tfvars_dir()}/terraform.tfvars"
      ]
    }

    extra_arguments "retry_lock" {
      commands = ["${get_terraform_commands_that_need_locking()}"]
      arguments = ["-lock-timeout=20m"]
    }
  }
}
```

What is particularly obnoxious is that I can't even exit the process normally.  It captures the ctl-C and just swallows it while it blocks on the lock:

```
Samuels-MBP:bastion sgendler$ terragrunt plan --terragrunt-source ../../../../../stem-infra//bastion/  
[terragrunt] [/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion] 2017/11/14 14:37:51 Running command: terraform --version
[terragrunt] 2017/11/14 14:37:51 Reading Terragrunt config file at /Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion/terraform.tfvars
[terragrunt] 2017/11/14 14:37:51 Cleaning up existing *.tf files in /var/folders/xr/t6gsrby97350k0r85qr7blzh0000gn/T/terragrunt/Fxxb1YsbNFq0VqlPDqwElrdbe-4/N7YR7JXFv_AHrQ_vUpC9GGTlLbM
[terragrunt] 2017/11/14 14:37:51 Downloading Terraform configurations from file:///Users/sgendler/src/stem/stem-infra into /var/folders/xr/t6gsrby97350k0r85qr7blzh0000gn/T/terragrunt/Fxxb1YsbNFq0VqlPDqwElrdbe-4/N7YR7JXFv_AHrQ_vUpC9GGTlLbM using terraform init
[terragrunt] [/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion] 2017/11/14 14:37:51 Backend s3 has not changed.
[terragrunt] [/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion] 2017/11/14 14:37:52 Running command: terraform init -backend-config=bucket=stem-terraform-state-bucket -backend-config=key=aws2/us-east-1/_global/bastion/terraform.tfstate -backend-config=region=us-west-2 -backend-config=encrypt=true -backend-config=dynamodb_table=terraform-lock-table -lock-timeout=20m -from-module=file:///Users/sgendler/src/stem/stem-infra /var/folders/xr/t6gsrby97350k0r85qr7blzh0000gn/T/terragrunt/Fxxb1YsbNFq0VqlPDqwElrdbe-4/N7YR7JXFv_AHrQ_vUpC9GGTlLbM
Copying configuration from "file:///Users/sgendler/src/stem/stem-infra"...
Terraform initialized in an empty directory!

The directory has no Terraform configuration files. You may begin working
with Terraform immediately by creating Terraform configuration files.
[terragrunt] 2017/11/14 14:37:52 Copying files from /Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion into /var/folders/xr/t6gsrby97350k0r85qr7blzh0000gn/T/terragrunt/Fxxb1YsbNFq0VqlPDqwElrdbe-4/N7YR7JXFv_AHrQ_vUpC9GGTlLbM/bastion
[terragrunt] 2017/11/14 14:37:52 Setting working directory to /var/folders/xr/t6gsrby97350k0r85qr7blzh0000gn/T/terragrunt/Fxxb1YsbNFq0VqlPDqwElrdbe-4/N7YR7JXFv_AHrQ_vUpC9GGTlLbM/bastion
[terragrunt] 2017/11/14 14:37:52 Backend s3 has not changed.
[terragrunt] 2017/11/14 14:37:52 Running command: terraform plan -var-file=/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion/../../../account.tfvars -var-file=/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion/../../region.tfvars -var-file=/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion/../env.tfvars -var-file=/Users/sgendler/src/stem/stem-envs/aws2/us-east-1/_global/bastion/terraform.tfvars -lock-timeout=20m
Acquiring state lock. This may take a few moments...
^C[terragrunt] 2017/11/14 14:41:35 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:36 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:37 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:37 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:37 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:40 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:40 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:40 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:40 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:40 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:40 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:41 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:41 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:41 Forward signal interrupt to terraform.
^C[terragrunt] 2017/11/14 14:41:41 Forward signal interrupt to terraform.
```

**Comment by sgendler-stem** [comment]
If I reduce the timeout, I get the following when it finally times-out

```
Acquiring state lock. This may take a few moments...

Error: Error locking state: Error acquiring the state lock: ConditionalCheckFailedException: The conditional request failed
	status code: 400, request id: 61TQCU8QIELPMR32HASGNS1M0NVV4KQNSO5AEMVJF66Q9ASUAAJG
Lock Info:
  ID:        0f001320-6483-073c-e490-715a2a048ab9
  Path:      stem-terraform-state-bucket/aws2/us-east-1/_global/bastion/terraform.tfstate
  Operation: OperationTypeApply
  Who:       sgendler@Samuels-MBP
  Version:   0.10.8
  Created:   2017-11-13 23:59:54.146334765 +0000 UTC
  Info:      


Terraform acquires a state lock to protect the state from being written
by multiple users at the same time. Please resolve the issue above and try
again. For most commands, you can disable locking with the "-lock=false"
flag, but this is not recommended.


[terragrunt] 2017/11/14 14:51:24 exit status 1
```

**Comment by sgendler-stem** [comment]
So how do I actually remove the lock? I ran commands with -lock=false, which works, but it doesn't remove the perma-locked lock.  I can't just have that entire module blow up without ignoring the lock forevermore.

**Comment by brikis98** [comment]
Terragrunt does not do locking. This is Terraform itself. My best guess is that if you lost connectivity during some previous `apply`, that something with the lock got messed up, and stale data is sitting around in your DynamoDB table. You can try cleaning it out by hand. If you can't get it resolved, please file a bug with Terraform itself, as Terragrunt just forwards your lock settings to it directly!

**Comment by sgendler-stem** [comment]
Sorry, I actually thought I was on the terraform github.  I must have grabbed the wrong tab.
